semeio/jobs/auto_correlate_observations/config.py

```diff
@@ -0,0 +1,315 @@
+import collections
+import configsuite as cs
+import copy
+import fnmatch
+
+
+_SCALING = "scaling"
+_THRESHOLD = "threshold"
+_STD_CUTOFF = "std_cutoff"
+_ALPHA = "alpha"
+_CLUSTERING = "clustering"
+_METHOD = "method"
+_SPEARMAN_CORRELATION = "spearman_correlation"
+_FCLUSTER = "fcluster"
+_T = "t"
+_CRITERION = "criterion"
+_INCONSISTENT = "inconsistent"
+_DISTANCE = "distance"
+_MAXCLUST = "maxclust"
+_MONOCRIT = "monocrit"
+_MAXCLUST_MONOCRIT = "maxclust_monocrit"
+_DEPTH = "depth"
+_LINKAGE = "linkage"
+_SINGLE = "single"
+_COMPLETE = "complete"
+_AVERAGE = "average"
+_WEIGHTED = "weighted"
+_CENTROID = "centroid"
+_WARD = "ward"
+_METRIC = "metric"
+_EUCLIDEAN = "euclidean"
+_METRICS = (
+    "braycurtis",
+    "canberra",
+    "chebyshev",
+    "cityblock",
+    "correlation",
+    "cosine",
+    "dice",
+    _EUCLIDEAN,
+    "hamming",
+    "jaccard",
+    "jensenshannon",
+    "kulsinski",
+    "mahalanobis",
+    "matching",
+    "minkowski",
+    "rogerstanimoto",
+    "russellrao",
+    "seuclidean",
+    "sokalmichener",
+    "sokalsneath",
+    "sqeuclidean",
+    "yule",
+)
+_OBSERVATIONS = "observations"
+
+
+def _bounds_validator(
+    lower=None, lower_inclusive=True, upper=None, upper_inclusive=True
+):
+    msg = "Value must be within the interval {}, {}".format(
+        "<"
+        if lower is None
+        else "{}{}".format("[" if lower_inclusive else "(", lower,),
+        ">"
+        if upper is None
+        else "{}{}".format(upper, "]" if upper_inclusive else ")",),
+    )
+
+    @cs.validator_msg(msg)
+    def validator(value):
+        lower_valid = lower is None or (
+            lower <= value if lower_inclusive else lower < value
+        )
+        upper_valid = upper is None or (
+            value <= upper if upper_inclusive else value < upper
+        )
+        return lower_valid and upper_valid
+
+    return validator
+
+
+def _one_of(*valid_values):
+    msg = "Value must be one of: ({})".format(", ".join(map(str, valid_values)))
+
+    @cs.validator_msg(msg)
+    def validator(value):
+        return value in valid_values
+
+    return validator
+
+
+@cs.validator_msg("t must be an integer if a max cluster criteria is choosen")
+def _t_is_int_if_maxclust(spearman_config):
+    if spearman_config[_CRITERION] in (_MAXCLUST, _MAXCLUST_MONOCRIT):
+        return isinstance(spearman_config[_T], int)
```

Comment:
what's "t"? Can we replace it with something more descriptive?

Reply:
_CLUSTER_THRESHOLD = "t" ?

Reply:
Same argument as above. The `scipy` docs refer to the argument as `t`. I'll write up the explanations and then there will be text explaining the value in the docs. Again, we can change the name and/or behaviour, but then we are not purely exposing the `scipy` parameters anymore. Which is also fine 🤷